ayon_api/entity_hub.py

```diff
@@ -1485,6 +1486,704 @@ def fill_children_ids(self, children_ids):
 
 
+class ProjectStatus:
+    """Project status class.
+
+    Args:
+        name (str): Name of the status. e.g. 'In progress'
+        short_name (Optional[str]): Short name of the status. e.g. 'IP'
+        state (Optional[Literal[not_started, in_progress, done, blocked]]): A
+            state of the status.
+        icon (Optional[str]): Icon of the status. e.g. 'play_arrow'.
+        color (Optional[str]): Color of the status. e.g. '#eeeeee'.
+        index (Optional[int]): Index of the status.
+        project_statuses (Optional[_ProjectStatuses]): Project statuses
+            wrapper.
+    """
+
+    valid_states = ("not_started", "in_progress", "done", "blocked")
+    color_regex = re.compile(r"#([a-f0-9]{6})$")
+    default_state = "in_progress"
+    default_color = "#eeeeee"
+
+    def __init__(
+        self,
+        name,
+        short_name=None,
+        state=None,
+        icon=None,
+        color=None,
+        index=None,
+        project_statuses=None,
+    ):
+        short_name = short_name or ""
+        icon = icon or ""
+        state = state or self.default_state
+        color = color or self.default_color
+        self._name = name
+        self._short_name = short_name
+        self._icon = icon
+        self._slugified_name = None
+        self._state = None
+        self._color = None
+        self.set_state(state)
+        self.set_color(color)
+
+        self._original_name = name
+        self._original_short_name = short_name
+        self._original_icon = icon
+        self._original_state = state
+        self._original_color = color
+        self._original_index = index
+
+        self._index = index
+        self._project_statuses = project_statuses
+        self._is_new = index is None or project_statuses is None
+
+    def __str__(self):
+        short_name = ""
+        if self.short_name:
+            short_name = "({})".format(self.short_name)
+        return "<{} {}{}>".format(
+            self.__class__.__name__, self.name, short_name
+        )
+
+    def __repr__(self):
+        return str(self)
+
+    def __getitem__(self, key):
+        if key in {
+            "name", "short_name", "icon", "state", "color", "slugified_name"
+        }:
+            return getattr(self, key)
+        raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        if key in {"name", "short_name", "icon", "state", "color"}:
+            return setattr(self, key, value)
+        raise KeyError(key)
+
+    def lock(self):
+        """Lock statuse.
+
+        Changes were commited and current values are now the original values.
+        """
+
+        self._is_new = False
+        self._original_name = self.name
+        self._original_short_name = self.short_name
+        self._original_icon = self.icon
+        self._original_state = self.state
+        self._original_color = self.color
+        self._original_index = self.index
+
+    @staticmethod
+    def slugify_name(name):
```

Comment:
We could use the `__eq__` here to compare two statuses or default compare the name.

Reply:
I don't thinks it is appropriate for comparison of the objects... They can have different short name, state or other values. And is question if should eq use slugified name, or standard name. Different usecases for different needs and I would say it's better to not implement it than try to give false-positive.